source/common/router/router.cc

```diff
@@ -521,6 +521,10 @@ void Filter::onRequestComplete() {
       response_timeout_ = dispatcher.createTimer([this]() -> void { onResponseTimeout(); });
       response_timeout_->enableTimer(timeout_.global_timeout_);
     }
+
+    if (pending_per_try_timeout_) {
```

Comment:
this might be better as a for loop over upstream requests to check their pending_per_try_timeout_ flag